src/components/datagrid/body/data_grid_cell.tsx

```diff
@@ -96,7 +92,18 @@ const EuiDataGridCellContent: FunctionComponent<
           />
         </div>
         <EuiScreenReaderOnly>
-          <p>{positionText}</p>
+          <p>
+            {'- '}
+            <EuiI18n
+              token="euiDataGridCell.position"
+              default="Row {row}, Column {col} ({columnId})"
+              values={{
+                row: ariaRowIndex,
+                col: colIndex + 1,
+                columnId: column?.displayAsText || rest.columnId,
+              }}
+            />
+          </p>
```

Comment:
Adding the `(column Name/ID)` in parenthesis helps contextualize the cell more, similar to default aria table behavior. I changed the `-` and `:` punctuation for a more natural sounding voice flow - let me know what you think!

### Before:
(Note: GH's video player appears to start muted, be sure to unmute it)

https://user-images.githubusercontent.com/549407/177643102-5aa04c55-7249-48e4-917a-21f8d1a32fb4.mp4

### After
(Note: GH's video player appears to start muted, be sure to unmute it)

https://user-images.githubusercontent.com/549407/177643113-e80e5ff7-8b53-4f7f-aaf8-daf70e7bc22c.mp4

Reply:
I've been listening to this read back all morning and thinking of it in terms of data importance priority. (I totally made that up). I swapped the `Column Name/ID` ahead of the row and column number combo. Putting myself in the place of a user digging through a column of data, I wanted to hear the data first, then the label, then my position in the grid. What do you think about swapping line 99 to be

```diff
! data_grid_cell#L99

<EuiScreenReaderOnly>
  <p>
    {'- '}
      <EuiI18n
        token="euiDataGridCell.position"
-       default="Row {row}, Column {col} ({columnId})"
+       default="{columnId}. Row {row}, Column {col}."
        values={{
        row: ariaRowIndex,
        col: colIndex + 1,
        columnId: column?.displayAsText || rest.columnId,
      }}
    />
  </p>
</EuiScreenReaderOnly>
```

Reply:
@constancecchen Good call. Let's keep the column name and index together. 

Your point about some Kibana column titles being technical and long is spot on. It matters, but how much is a sliding scale. I'm floating the idea elsewhere, and feel the benefits outweigh the increased cognitive load. If research proves out otherwise, we can always revisit this solution.

Reply:
Gotcha! So just to confirm, do you want to move stay with the current copy, or do you want to change it to:

```suggestion
          <p>
            {'- '}
            <EuiI18n
              token="euiDataGridCell.position"
              default="{columnId}, column {col}, row {row}"
              values={{
                row: ariaRowIndex,
                col: colIndex + 1,
                columnId: column?.displayAsText || rest.columnId,
              }}
            />
          </p>
```

?

Reply:
Let's change it to the `default="{columnId}, column {col}, row {row}"` you proposed in the green addition.

My apologies, comment clarity is a work in progress.

Reply:
https://github.com/elastic/eui/pull/6033/commits/bada576888d4a5c6416f6dab6399aca6d872b6fb